lib/internal/tty.js

```diff
@@ -72,11 +72,49 @@ const TERM_ENVS_REG_EXP = [
   /^vt100/
 ];
 
+function warnOnDeactivatedColors(env) {
+  let name;
+  if (env.NODE_DISABLE_COLORS !== undefined)
+    name = 'NODE_DISABLE_COLORS';
+  if (env.NO_COLOR !== undefined)
+    name = 'NO_COLOR';
+
+  if (name !== undefined) {
+    process.emitWarning(
+      `The '${name}' env is ignored due to the 'FORCE_COLOR' env being set.`,
+      'Warning'
+    );
+  }
+}
+
 // The `getColorDepth` API got inspired by multiple sources such as
 // https://github.com/chalk/supports-color,
 // https://github.com/isaacs/color-support.
 function getColorDepth(env = process.env) {
-  if (env.NODE_DISABLE_COLORS || env.TERM === 'dumb') {
+  // Use level 0-3 to support the same levels as `chalk` does. This is done for
+  // consistency throughout the ecosystem.
+  if (env.FORCE_COLOR !== undefined) {
+    switch (env.FORCE_COLOR) {
+      case '':
+      case '1':
+      case 'true':
+        warnOnDeactivatedColors(env);
```

Comment:
Place this above the switch?

Reply:
When moved above the switch, I'll have to exclude a lot of values. That's why it's inside.